### ISSUE TITLE
get/getRows doesn't work for column vector

### ISSUE THREAD

**Issue by allanj** [deeplearning4j/nd4j#1364]
As an example, I copied the code from my post in gist: https://gist.github.com/allanj/2a42d9e0fdc0cc2f8e6e90337b4507a5
I use 0.6.0 version

```
import org.nd4j.linalg.api.ndarray.INDArray;
import org.nd4j.linalg.factory.Nd4j;

public class Test {


   public static void main(String... args){
        INDArray mat1 = Nd4j.create(new double[][]{{3,5},{2,7},{5,4},{6,3}});
        System.out.println("matrix 1:"+mat1.getRows(new int[]{1, 3}));

        INDArray mat2 = Nd4j.create(new double[][]{{3},{2},{5},{6}});
        System.out.println("matrix 2:"+mat2.getRows(new int[]{1, 3}));
   }
}
```

The output and stack trace is:
```
[main] INFO org.nd4j.nativeblas.NativeOps - Number of threads used for NativeOps: 4
Unable to guess runtime. Please set OMP_NUM_THREADS or equivalent manually.
[main] INFO org.nd4j.nativeblas.Nd4jBlas - Number of threads used for BLAS: 4
[main] INFO org.reflections.Reflections - Reflections took 210 ms to scan 10 urls, producing 121 keys and 415 values
[main] INFO org.reflections.Reflections - Reflections took 80 ms to scan 10 urls, producing 121 keys and 415 values

matrix 1:[[2.00, 7.00],   
[6.00, 3.00]]

Exception in thread "main" java.lang.UnsupportedOperationException: Illegal combination of indexes for vector   
at org.nd4j.linalg.indexing.ShapeOffsetResolution.tryShortCircuit(ShapeOffsetResolution.java:135)   
at org.nd4j.linalg.indexing.ShapeOffsetResolution.exec(ShapeOffsetResolution.java:325)   
at org.nd4j.linalg.api.ndarray.BaseNDArray.get(BaseNDArray.java:3767)  
at org.nd4j.linalg.api.ndarray.BaseNDArray.getRows(BaseNDArray.java:3750)    
```

I also tried `get(new SpecifidIndex(1,3),NDArrayIndex.all())` for my column vector. It still doesn't work. But if I use `get(NDArrayIndex.all(),new SpecifidIndex(1,3))` for row vector. It works. 

My current solution is just to transpose the vector.

**Comment by allanj** [comment]
Does anyone has this issue or should be my own problem?

**Comment by AlexDBlack** [comment]
Confirmed as an issue for 0.6.0, and confirmed fixed for 0.7.0.
